examples/gazebo_scene_viewer/CMakeLists.txt

```diff
@@ -17,8 +17,12 @@ if (NOT APPLE)
   find_package(GLEW REQUIRED)
   include_directories(SYSTEM ${GLEW_INCLUDE_DIRS})
   link_directories(${GLEW_LIBRARY_DIRS})
+  set(STD_CXX_FS_LIBRARIES "stdc++fs")
 endif()
 
+find_package(bullet REQUIRED)
+link_directories(${BULLET_LIBRARY_DIRS})
```

Comment:
what's the reason for finding and linking against bullet?

Reply:
Without bulIet I see a link error against gazebo11:

```
ld: library not found for -lBulletSoftBody
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [gazebo_scene_viewer2_demo] Error 1
make[1]: *** [CMakeFiles/gazebo_scene_viewer2_demo.dir/all] Error 2
make: *** [all] Error 2
```
I am using gazebo11 built from source with the physics libraries, including bullet, installed using brew:

```bash
$ brew info bullet
bullet: stable 3.08 (bottled), HEAD
Physics SDK
https://bulletphysics.org/
/usr/local/Cellar/bullet/3.08_2 (1,341 files, 25.0MB) *
  Poured from bottle on 2021-01-18 at 14:08:30
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/bullet.rb
License: Zlib
==> Dependencies
Build: cmake ✔, pkg-config ✔, python@3.9 ✔
==> Options
--HEAD
	Install HEAD version
```

When gazebo is built it finds and links against bullet if it is available. So bullet seems to be transitively required when the examples link to gazebo. I was surprised to see this as well as the demo does not directly include any physics headers.

Here is the full output from cmake with the find_package(bullet) commented out (debug config). It appears `/usr/local/lib/cmake/gazebo/gazebo-config.cmake` is looking for physics libraries:

```bash
cmake ../.. -DCMAKE_INSTALL_PREFIX=/usr/local/Cellar/ignition_debug/2.0 -DCMAKE_BUILD_TYPE=Debug
-- Looking for ignition-rendering4 -- found version 4.5.0
-- Searching for dependencies of ignition-rendering4
-- Looking for ignition-math6 -- found version 6.8.0~pre2
-- Searching for dependencies of ignition-math6
-- Looking for ignition-common3 -- found version 3.10.1
-- Searching for dependencies of ignition-common3
-- Looking for dlfcn.h - found
-- Looking for libdl - found
-- Searching for <ignition-common3> component [graphics]
-- Looking for ignition-common3-graphics -- found version 3.10.1
-- Searching for dependencies of ignition-common3-graphics
-- Looking for ignition-math6 -- found version 6.8.0~pre2
-- Searching for <ignition-common3> component [events]
-- Looking for ignition-common3-events -- found version 3.10.1
-- Searching for dependencies of ignition-common3-events
-- Looking for ignition-math6 -- found version 6.8.0~pre2
-- Looking for ignition-plugin1 -- found version 1.1.0
-- Searching for dependencies of ignition-plugin1
-- Searching for <ignition-plugin1> component [all]
-- Looking for all libraries of ignition-plugin1 -- found version 1.1.0
-- Looking for ignition-plugin1 -- found version 1.1.0
-- Looking for ignition-plugin1-loader -- found version 1.1.0
-- Searching for dependencies of ignition-plugin1-loader
-- Looking for ignition-plugin1-register -- found version 1.1.0
-- Searching for dependencies of ignition-plugin1-register
CMake Warning (dev) at /usr/local/Cellar/cmake/3.19.4/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:426 (message):
  The package name passed to `find_package_handle_standard_args` (PkgConfig)
  does not match the name of the calling package (gazebo).  This can lead to
  problems in calling code that expects `find_package` result variables
  (e.g., `_FOUND`) to follow a certain pattern.
Call Stack (most recent call first):
  /usr/local/Cellar/cmake/3.19.4/share/cmake/Modules/FindPkgConfig.cmake:67 (find_package_handle_standard_args)
  /usr/local/lib/cmake/gazebo/gazebo-config.cmake:31 (include)
  CMakeLists.txt:4 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Found DART: /usr/local/include (Required is at least version "6.6") found components: dart 
-- Found Protobuf: /usr/local/lib/libprotobuf.dylib (found version "3.14.0") 
-- Looking for ignition-math6 -- found version 6.8.0~pre2
-- Looking for OGRE...
-- Found Ogre Ghadamon (1.9.0)
-- Found OGRE: optimized;/usr/local/Cellar/ogre1.9/1.9.0/lib/libOgreMain.dylib;debug;/usr/local/Cellar/ogre1.9/1.9.0/lib/libOgreMain.dylib
-- Looking for OGRE_Paging...
-- Found OGRE_Paging: optimized;/usr/local/Cellar/ogre1.9/1.9.0/lib/libOgrePaging.dylib;debug;/usr/local/Cellar/ogre1.9/1.9.0/lib/libOgrePaging.dylib
-- Looking for OGRE_Terrain...
-- Found OGRE_Terrain: optimized;/usr/local/Cellar/ogre1.9/1.9.0/lib/libOgreTerrain.dylib;debug;/usr/local/Cellar/ogre1.9/1.9.0/lib/libOgreTerrain.dylib
-- Looking for OGRE_Property...
-- Found OGRE_Property: optimized;/usr/local/Cellar/ogre1.9/1.9.0/lib/libOgreProperty.dylib;debug;/usr/local/Cellar/ogre1.9/1.9.0/lib/libOgreProperty.dylib
-- Looking for OGRE_RTShaderSystem...
-- Found OGRE_RTShaderSystem: optimized;/usr/local/Cellar/ogre1.9/1.9.0/lib/libOgreRTShaderSystem.dylib;debug;/usr/local/Cellar/ogre1.9/1.9.0/lib/libOgreRTShaderSystem.dylib
-- Looking for OGRE_Volume...
-- Found OGRE_Volume: optimized;/usr/local/Cellar/ogre1.9/1.9.0/lib/libOgreVolume.dylib;debug;/usr/local/Cellar/ogre1.9/1.9.0/lib/libOgreVolume.dylib
-- Looking for OGRE_Overlay...
-- Found OGRE_Overlay: optimized;/usr/local/Cellar/ogre1.9/1.9.0/lib/libOgreOverlay.dylib;debug;/usr/local/Cellar/ogre1.9/1.9.0/lib/libOgreOverlay.dylib
-- Looking for ignition-math6 -- found version 6.8.0~pre2
-- Looking for ignition-transport8 -- found version 8.1.0
-- Searching for dependencies of ignition-transport8
-- Found Protobuf: /usr/local/lib/libprotobuf.dylib (found suitable version "3.14.0", minimum required is "3") 
-- Config-file not installed for ZeroMQ -- checking for pkg-config
-- Checking for module 'libzmq >= 4'
--   Found libzmq , version 4.3.4
-- Looking for ignition-msgs5 -- found version 5.3.0
-- Searching for dependencies of ignition-msgs5
-- Looking for ignition-math6 -- found version 6.8.0~pre2
-- Checking for module 'tinyxml2'
--   Found tinyxml2, version 8.0.0
-- Looking for ignition-msgs5 -- found version 5.3.0
-- Looking for ignition-common3 -- found version 3.10.1
-- Looking for ignition-fuel_tools4 -- found version 4.2.1
-- Searching for dependencies of ignition-fuel_tools4
-- Checking for module 'jsoncpp'
--   Found jsoncpp, version 1.9.4
-- Checking for module 'yaml-0.1'
--   Found yaml-0.1, version 0.2.5
-- Checking for module 'libzip'
--   Found libzip, version 1.7.3
-- Looking for ignition-common3 -- found version 3.10.1
-- Looking for ignition-msgs5 -- found version 5.3.0
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/rhys/Code/robotics/gazebo/ign_ws/src/ign-rendering/examples/gazebo_scene_viewer/build/debug
```

I have set the install target to `/usr/local/Cellar/ignition_debug/2.0` which is then linked using brew into `/usr/local` when needed. This keeps the debug and release builds separated but makes them easy to deploy where the system can find them. It also seems to resolve DYLD_LIBRARY_PATH issues I am seeing when using colcon to build into a local folder (I do not disable system integrity protection on macOS).

Reply:
Here's the [gazebo-config.cmake.in](https://github.com/osrf/gazebo/blob/gazebo11/cmake/gazebo-config.cmake.in#L94) file that is used to generate `gazebo-config.cmake`. Looking at your output, it seems like it's not trying to find bullet at all, which means some of the `if` checks for bullet evaluate to false. Do you have `pkg-config`? It could be [failing here](https://github.com/osrf/gazebo/blob/gazebo11/cmake/gazebo-config.cmake.in#L94)

Reply:
Yes, I have `pkg-config`  installed with brew. Turns out that ` gazebo-config.cmake.in` is appending the Bullet libraries to `GAZEBO_LIBRARIES` but `cmake` is not finding the libraries in `/usr/local/Cellar/bullet/3.08_2/lib`, but will find them if I add `link_libraries(/usr/local/lib)` to `CMakeLists.txt`.

I'll put this down to something odd with my setup and revert the changes to `CMakeLists.txt`.

Reply:
ok thanks for looking into it. I'll mark this as resolved. Changes look good. Thanks for the patch